lib/rpmsg/rpmsg_virtio.c

```diff
@@ -109,14 +111,16 @@ static int rpmsg_virtio_enqueue_buffer(struct rpmsg_virtio_device *rvdev,
 
 		/* Initialize buffer node */
 		vqbuf.buf = buffer;
-		vqbuf.len = len;
+		vqbuf.len = len ? len : RPMSG_BUFFER_SIZE;
```

Comment:
Just in case of non-zero copy, len "0" indicates error, prefer to set the len in this function's caller. And only limit to use default len in the zero copy functions.

Reply:
Ok.